### ISSUE TITLE
Error while building my app

### ISSUE THREAD

**Issue by Shenhe0310** [brillout/vite-plugin-ssr#771]
### Description

Hi i got an error while building my app with vite-plugin-ssr. Im using Vue3 and vue router with vite. Will that be a problem? Im learning with progamming so if you need more info about this error tell me!

my vite.config.js file:
import { fileURLToPath, URL } from "node:url";

import { createRequire } from "node:module";
const require = createRequire(import.meta.url);

import { defineConfig } from "vite";
import vue from "@vitejs/plugin-vue";

// import ssr from "vite-plugin-ssr/plugin";
// import { ViteSSR } from "vite-plugin-ssr/server";
import { ssr } from "vite-plugin-ssr/plugin";

import ckeditor5 from "@ckeditor/vite-plugin-ckeditor5";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    vue(),
    ckeditor5({ theme: require.resolve("@ckeditor/ckeditor5-theme-lark") }),
    ssr({ prerender: true }),
  ],

  resolve: {
    alias: {
      "@": fileURLToPath(new URL("./src", import.meta.url)),
    },
  },

  build: {
    // outDir: "dist",
    // ssrManifest: true,
    // emptyOutDir: true,
    rollupOptions: {
      input: "src/main.js",
    },
  },
});

this is all i changed when using [vite-plugin-ssr]

### Error Message + Error Stack

```shell
[vite-plugin-ssr@0.4.111][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Go to https://github.com/brillout/vite-plugin-ssr/issues/new and copy-paste this error. (The error's stack trace is usually enough 
to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as it makes vite-plugin-ssr more robust. Debug info (this is for the vite-plugin-ssr maintainers; you can ignore this): Wrongly formatted path: C:\myproject\Albedo\src\main.js
error during build:
Error: [vite-plugin-ssr@0.4.111][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Go to https://github.com/brillout/vite-plugin-ssr/issues/new and copy-paste this error. (The error's stack trace is usually 
enough to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as it makes vite-plugin-ssr more robust. Debug info (this is for the vite-plugin-ssr maintainers; you can ignore this): Wrongly formatted path: C:\myproject\Albedo\src\main.js
    at assertPosixPath (C:\myproject\Albedo\node_modules\vite-plugin-ssr\dist\cjs\utils\filesystemPathHandling.js:18:25)
    at getScriptFileName (C:\myproject\Albedo\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\distFileNames.js:65:37)
    at rollupOutput.entryFileNames (C:\myproject\Albedo\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\distFileNames.js:24:66)
    at Chunk.getPreliminaryFileName (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:15277:74)
    at reserveEntryChunksInBundle (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:16607:19)
    at renderChunks (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:16593:5)        
    at Bundle.generate (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:16807:19)    
    at async file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:25030:27
    at async catchUnfinishedHookActions (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:24133:20)
    at async build (file:///C:/myproject/Albedo/node_modules/vite/dist/node/chunks/dep-ca21228b.js:44670:22)  
    at async CAC.<anonymous> (file:///C:/myproject/Albedo/node_modules/vite/dist/node/cli.js:813:9)
```

**Comment by brillout** [comment]
If you replace the following at line `64` of `C:\myproject\Albedo\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\distFileNames.js`:

```diff
     const id = chunkInfo.facadeModuleId;
-    if (id)
-        (0, utils_1.assertPosixPath)(id);
+    if (id) {
+        id = (0, utils_1.toPosixPath)(id);
+        (0, utils_1.assertPosixPath)(id);
+    }
```

Does it work then?

**Comment by Shenhe0310** [comment]
I did what you said and i got this error:

Assignment to constant variable.
error during build:
TypeError: Assignment to constant variable.
    at getScriptFileName (C:\myproject\Albedo\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\distFileNames.js:67:12)
    at rollupOutput.entryFileNames (C:\myproject\Albedo\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\distFileNames.js:24:66)
    at Chunk.getPreliminaryFileName (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:15277:74) 
    at reserveEntryChunksInBundle (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:16607:19)   
    at renderChunks (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:16593:5)
    at Bundle.generate (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:16807:19)
    at async file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:25030:27
    at async catchUnfinishedHookActions (file:///C:/myproject/Albedo/node_modules/rollup/dist/es/shared/rollup.js:24133:20)
    at async build (file:///C:/myproject/Albedo/node_modules/vite/dist/node/chunks/dep-ca21228b.js:44670:22)
    at async CAC.<anonymous> (file:///C:/myproject/Albedo/node_modules/vite/dist/node/cli.js:813:9)

**Comment by brillout** [comment]
Change `const id` to `let id`.

**Comment by Shenhe0310** [comment]
It worked!

vite v4.1.4 building for production...
✓ 1408 modules transformed.
computing gzip size (1)...vite v4.1.4 building SSR bundle for production...
dist/client/vite-plugin-ssr.json                 0.16 kB
dist/client/manifest.json                        0.28 kB
dist/client/assets/static/main.69198351.css    307.41 kB │ gzip:  48.04 kB
dist/client/assets/src/main.js.d2d81080.js   2,184.55 kB │ gzip: 613.51 kB

(!) Some chunks are larger than 500 kBs after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
transforming (29) virtual:vite-plugin-ssr:importUserCode:server"updateDoc", "where", "limit" and "getDocs" are imported from external module "firebase/firestore" but never used in "src/assets/JS files/ckimageupload.js", "src/main.js", "src/App.vue", "src/views/TestPage/testfire.vue", "src/views/TestPage/TestPage.vue", "src/views/TestPage/TestPage5.vue" and "src/main.js".
✓ 47 modules transformed.
dist/server/package.json      0.02 kB
dist/server/importBuild.cjs   0.35 kB
dist/server/importBuild.mjs   0.23 kB
dist/server/pageFiles.mjs     1.77 kB
dist/server/src/main.js.mjs  81.87 kB
vite-plugin-ssr v0.4.111 pre-rendering HTML...
✓ 0 HTML documents pre-rendered.

Thanks a lot😍

**Comment by brillout** [comment]
Released as `0.4.112`.

In case you think you can convince your employer: https://github.com/sponsors/brillout.
